From 2b1f2cab1e5c2674ae0e5747cf41d4ebc48c1182 Mon Sep 17 00:00:00 2001
From: lunatickochiya <125438787@qq.com>
Date: Mon, 7 Apr 2025 20:02:13 +0800
Subject: [PATCH] mediatek:fix cmcc rax3000m custom uboot mod Revert "Auxiliary
 commit to revert individual files from
 553d21865fe2f3bc1ace1c309920f1f4ace1df97"

This reverts commit 62f34b287e1a9f96cfe4b24e1550f72875cc6c89.
mediatek:fix cmcc rax3000m emmc custom uboot mod
---
 package/boot/uboot-envtools/files/mediatek_filogic          | 3 +++
 .../mediatek/dts/mt7981b-cmcc-rax3000m-emmc-ubootmod.dts    | 3 +--
 .../mediatek/dts/mt7981b-cmcc-rax3000m-nand-ubootmod.dts    | 4 ++++
 .../base-files/etc/hotplug.d/firmware/11-mt76-caldata       | 4 +---
 .../filogic/base-files/lib/preinit/10_fix_eth_mac.sh        | 6 ++++++
 5 files changed, 15 insertions(+), 5 deletions(-)

diff --git a/package/boot/uboot-envtools/files/mediatek_filogic b/package/boot/uboot-envtools/files/mediatek_filogic
index bb40e1ad3b..e69ca79009 100644
--- a/package/boot/uboot-envtools/files/mediatek_filogic
+++ b/package/boot/uboot-envtools/files/mediatek_filogic
@@ -142,6 +142,9 @@ zyxel,ex5601-t0)
 zyxel,ex5700-telenor)
 	ubootenv_add_uci_config "/dev/ubootenv" "0x0" "0x4000" "0x4000" "1"
 	;;
+cmcc,rax3000m-emmc-ubootmod)
+	ubootenv_add_uci_config "/dev/mmcblk0p1" "0x0" "0x80000" "0x80000"
+	;;
 esac
 
 config_load ubootenv
diff --git a/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-emmc-ubootmod.dts b/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-emmc-ubootmod.dts
index 0008f0832a..c00875a243 100644
--- a/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-emmc-ubootmod.dts
+++ b/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-emmc-ubootmod.dts
@@ -8,9 +8,8 @@
 	compatible = "cmcc,rax3000m-emmc-ubootmod", "mediatek,mt7981";
 
 	chosen {
-		bootargs = "root=PARTLABEL=rootfs rootwait rootfstype=squashfs,f2fs";
+		bootargs-override = "root=PARTLABEL=rootfs rootwait rootfstype=squashfs,f2fs";
 	};
-
 };
 
 &mmc0 {
diff --git a/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-nand-ubootmod.dts b/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-nand-ubootmod.dts
index 5d4f5a77a7..a36135d532 100644
--- a/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-nand-ubootmod.dts
+++ b/target/linux/mediatek/dts/mt7981b-cmcc-rax3000m-nand-ubootmod.dts
@@ -12,6 +12,10 @@
 	};
 };
 
+&chosen {
+	bootargs-override = "";
+};
+
 &gmac0 {
 	nvmem-cells = <&macaddr_factory_2a 0>;
 	nvmem-cell-names = "mac-address";
diff --git a/target/linux/mediatek/filogic/base-files/etc/hotplug.d/firmware/11-mt76-caldata b/target/linux/mediatek/filogic/base-files/etc/hotplug.d/firmware/11-mt76-caldata
index 574cd8b02b..ac01ae8a99 100644
--- a/target/linux/mediatek/filogic/base-files/etc/hotplug.d/firmware/11-mt76-caldata
+++ b/target/linux/mediatek/filogic/base-files/etc/hotplug.d/firmware/11-mt76-caldata
@@ -9,12 +9,10 @@ board=$(board_name)
 case "$FIRMWARE" in
 "mediatek/mt7981_eeprom_mt7976_dbdc.bin")
 	case "$board" in
+	cmcc,rax3000m-emmc-ubootmod|\
 	ubnt,unifi-6-plus)
 		caldata_extract_mmc "factory" 0x0 0x1000
 		;;
-	cmcc,rax3000m-emmc-ubootmod)
-		caldata_extract_mmc "factory" 0x0 0x1000
-		;;
 	esac
 	;;
 "mediatek/mt7986_eeprom_mt7975_dual.bin")
diff --git a/target/linux/mediatek/filogic/base-files/lib/preinit/10_fix_eth_mac.sh b/target/linux/mediatek/filogic/base-files/lib/preinit/10_fix_eth_mac.sh
index 9d279898ac..b7f423dcab 100644
--- a/target/linux/mediatek/filogic/base-files/lib/preinit/10_fix_eth_mac.sh
+++ b/target/linux/mediatek/filogic/base-files/lib/preinit/10_fix_eth_mac.sh
@@ -32,6 +32,12 @@ preinit_set_mac_address() {
 		addr=$(get_mac_binary "/tmp/tp_data/default-mac" 0)
 		ip link set dev eth1 address "$(macaddr_add $addr 1)"
 		;;
+	cmcc,rax3000m-emmc-ubootmod)
+		addr=$(mmc_get_mac_binary factory 0x24)
+		ip link set dev eth0 address "$addr"
+		addr=$(mmc_get_mac_binary factory 0x2a)
+		ip link set dev eth1 address "$addr"
+		;;
 	*)
 		;;
 	esac
-- 
2.34.1

